src/snapshot_store/implementation.rs

```diff
@@ -42,29 +51,192 @@ impl SnapshotStoreImpl {
     ///
     /// Returns an error if configuration is invalid.
     pub fn new(config: Config) -> Result<Self, Error> {
+        // Get hostname for node_id
+        let node_id = hostname::get()
```

Comment:
I'm not sure using the hostname as the node if is ok. That seems inconsistent with the rest of the system (e.g. raft, libp2p). Its also unclear why SnapshotStore even needs a node id